integrations/kubecost/integration.py

```diff
@@ -56,125 +55,309 @@ def validate(cls, value: str) -> None:
-    )
-    share_cost: float = Field(
-        alias="shareCost",
-        default=0.0,
-        description="Floating-point value representing a monthly cost to share with the remaining non-idle, unshared allocations; e.g. 30.42 ($1.00/day == $30.42/month) for the query yesterday (1 day) will split and distribute exactly $1.00 across the allocations. Default is 0.0.",
-    )
-    filter_invoice_entity_ids: str | None = Field(
-        alias="filterInvoiceEntityIDs", description="Filter for account"
-    )
-    filter_account_ids: str | None = Field(
-        alias="filterAccountIDs", description="GCP only, filter for projectID"
-    )
-    filter_providers: str | None = Field(
-        alias="filterProviders", description="Filter for cloud service provider"
-    )
-    filter_label: str | None = Field(
-        alias="filterLabel",
-        description="Filter for a specific label. Does not support filtering for multiple labels at once.",
-    )
+class CloudCostV1ResourceConfig(ResourceConfig):
+    class CloudCostSelector(Selector):
+        window: (
+            Literal[
+                "today",
+                "week",
+                "month",
+                "yesterday",
+                "lastweek",
+                "lastmonth",
+                "30m",
+                "12h",
+                "7d",
+            ]
+            | DatePairField
+            | UnixtimePairField
+        ) = Field(default="today")
+        aggregate: AggregationField | None = Field(
+            description="Field by which to aggregate the results.",
+        )
+        filter_invoice_entity_ids: str | None = Field(
+            alias="filterInvoiceEntityIDs", description="Filter for account"
+        )
+        filter_account_ids: str | None = Field(
+            alias="filterAccountIDs", description="GCP only, filter for projectID"
+        )
+        filter_providers: str | None = Field(
+            alias="filterProviders", description="Filter for cloud service provider"
+        )
+        filter_label: str | None = Field(
+            alias="filterLabel",
+            description="Filter for a specific label. Does not support filtering for multiple labels at once.",
+        )
+        filter_services: str | None = Field(
+            alias="filterServices",
+            description="Comma-separated list of services to match; e.g. frontend-one,frontend-two will return results with either of those two services",
+        )
+
+    selector: CloudCostSelector
+    kind: Literal["cloud"]
 
 
-class KubecostResourceConfig(ResourceConfig):
+class CloudCostV2ResourceConfig(ResourceConfig):
+    class CloudCostSelector(Selector):
+        window: (
+            Literal[
+                "today",
+                "week",
+                "month",
+                "yesterday",
+                "lastweek",
+                "lastmonth",
+                "30m",
+                "12h",
+                "7d",
+            ]
+            | DatePairField
```

Comment:
lets move the "selector" class definition outside of the class of resourceConfig and only reference it, see how we do in datadog, azure etc..